src/Emanote/Pandoc/BuiltinFiltersSpec.hs

```diff
@@ -0,0 +1,48 @@
+module Emanote.Pandoc.BuiltinFiltersSpec where
+
+import Emanote.Pandoc.BuiltinFilters
+import Emanote.Pandoc.Markdown.Parser (parseMarkdown)
+import Hedgehog
+import Relude
+import Test.Hspec
+import Test.Hspec.Hedgehog (hedgehog)
+import Text.Pandoc.Definition (Inline (..))
+import Text.Pandoc.Walk (query)
+
+spec :: Spec
+spec =
+  do
+    describe "setExternalLinkicon" $ do
+      let run =
+            (>>=) . evalEither . fmap (query extractAttr . preparePandoc . snd) . parseMarkdown ""
+            where
+              extractAttr (Link (_, _, attrs) _ (_, _)) = snd <$> filter ((== "data-linkicon") . fst) attrs
```

Comment:
Should this be called `getDataLinkIconAttr` instead?

Reply:
Fixed.